src/Extend/SitesLinkedToMagentoStores.php

```diff
@@ -12,12 +13,15 @@ public function findByUrl($url)
             return $site;
         }
 
-        return parent::findByUrl($url);
+        return Cache::store('array')->rememberForever('rapidez.statamic.findByUrl-'.$url, fn() => parent::findByUrl($url));
     }
 
 
     public function findByMageRunCode($code)
     {
-        return collect($this->sites)->get($code);
+        if (!$code) {
+            return null;
+        }
+        return Cache::store('array')->rememberForever('rapidez.statamic.findByMageRunCode-'.$code, fn() => collect($this->sites)->get($code));
```

Comment:
Sites should be a collection already, so `$this->sites->ge($code)` should do the same, as `Sites` is a singleton I think any caching here doesn't have any effect.

Reply:
With the amount of times this was executing the constant casting to array, and from array to collection the caching did have an effect.

However, if we make sure it's a collection (which it should be) we can skip the `collection -> array -> collection`
I've updated the code